hello-world.md

```diff
@@ -6,11 +6,11 @@
 
 ## Specification
 
-The `Hello World!` program will greet me, the caller.
+Write a `Hello World!` function that takes a single input, the name of someone to greet, and returns an appropriate greeting.
```

Comment:
Ah, that's right, need to deal with multiple arities in a way that makes sense. Getting harder and harder to write something to write something generic over all languages!

Thinking through some possibilities:

* "Write a `Hello World!` function that can take a single input" - This, I suppose, leaves open the possibility that it can take something else other than a single input, although this wording doesn't make it explicit what that something else is.
* "Write a `Hello World!` function that can take a single optional input" - possibly, though some languages won't allow optional inputs, so this sentence may get confusing for those languages.
* "Write a `Hello World!` function that can greet someone given their name" - perhaps just don't mention the function inputs at all?